### PR TITLE
python3Packages.lexilang: init at 1.0.1

### PR DESCRIPTION
**Pull request by Izorkin**
## Description of changes
Add lexilang python package.
Required to LibrTranslate package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/lexilang/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-QAKJZRc29GssiLX2q9oRr5SYfaTDDrlk7EELiMxWiSU";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  doCheck = false;
```
Comment (bjornfor) [inline pkgs/development/python-modules/lexilang/default.nix:21]: Please add a comment above doCheck, explaining why it's disabled?

Comment (Izorkin) [reply pkgs/development/python-modules/lexilang/default.nix:21]: I couldn't find out reason for error:
```
python3.11-LexiLang> Running phase: pytestCheckPhase
python3.11-LexiLang> Executing pytestCheckPhase
python3.11-LexiLang> ============================= test session starts ==============================
python3.11-LexiLang> platform linux -- Python 3.11.7, pytest-7.4.3, pluggy-1.3.0
python3.11-LexiLang> rootdir: /build/LexiLang-1.0.1
python3.11-LexiLang> collected 0 items
python3.11-LexiLang> ============================ no tests ran in 0.01s =============================
python3.11-LexiLang> /nix/store/v5irq7wvkr7kih0hhnch5nnv2dcq8c4f-stdenv-linux/setup: line 1557: pop_var_context: head of shell_variables not a function context
```

Comment (mweinelt) [reply pkgs/development/python-modules/lexilang/default.nix:21]: Collected 0 items means it didn't the tests. Check if they are in the distribution you're pulling. If not, please fetch from git.

Comment (Izorkin) [reply pkgs/development/python-modules/lexilang/default.nix:21]: It is possible that these files are required - https://github.com/LibreTranslate/LexiLang/tree/main/dictionaries
I just don't know how to download them separately yet.

Comment (mweinelt) [reply pkgs/development/python-modules/lexilang/default.nix:21]: https://github.com/LibreTranslate/LexiLang/blob/main/test.py

I think this is the only test. I'm not sure about the dictionary files linked, but they're also not part of the sdist.

```nix
checkPhase = ''
  runHook preCheck
  ${python.interpreter} test.py
  runHook postCheck
'';

Comment (Izorkin) [reply pkgs/development/python-modules/lexilang/default.nix:21]: Thanks, fixed!

File: pkgs/development/python-modules/lexilang/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+    ${python.interpreter} test.py
+    runHook postCheck
+  '';
+
+  meta = with lib; {
+    description = "Simple, fast dictionary-based language detector for short texts.";
```
Comment (bjornfor) [inline pkgs/development/python-modules/lexilang/default.nix:36]: meta.description should not end with a dot.

Comment (Izorkin) [reply pkgs/development/python-modules/lexilang/default.nix]: Fixed.

**Review by bjornfor — APPROVED** [review]

File: pkgs/development/python-modules/lexilang/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, setuptools
+, pytestCheckHook
+, python
+}:
+
+buildPythonPackage rec {
+  pname = "LexiLang";
```
Comment (mweinelt) [inline pkgs/development/python-modules/lexilang/default.nix:10]:
```suggestion
  pname = "lexilang";
```

We normalize the pname in python packages.

File: pkgs/development/python-modules/lexilang/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  version = "1.0.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "LibreTranslate";
+    repo = pname;
```
Comment (mweinelt) [inline pkgs/development/python-modules/lexilang/default.nix:16]:
```suggestion
    repo = "LexiLang";
```

File: pkgs/development/python-modules/lexilang/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  checkPhase = ''
+    runHook preCheck
+    ${python.interpreter} test.py
+    runHook postCheck
+  '';
+
+  meta = with lib; {
+    description = "Simple, fast dictionary-based language detector for short texts";
+    homepage = "https://pypi.org/project/LexiLang/";
```
Comment (mweinelt) [inline pkgs/development/python-modules/lexilang/default.nix:37]:
```suggestion
    homepage = "https://github.com/LibreTranslate/LexiLang";
```

PyPi is not a homepage.

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/lexilang/default.nix (hunk 5 above)
Comment (Izorkin) [reply pkgs/development/python-modules/lexilang/default.nix]: Fixed.

**Review by mweinelt — APPROVED** [review]

**Comment by Izorkin** [comment]
Thanks!

**Pull request merged**